GVRf/Framework/jni/objects/hybrid_object_jni.cpp

```diff
@@ -28,7 +28,9 @@ Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,
 JNIEXPORT void JNICALL
 Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,
         jobject obj, jlong jhybrid_object) {
-    delete reinterpret_cast<HybridObject*>(jhybrid_object);
+    HybridObject* hybrid_object = reinterpret_cast<HybridObject*>(jhybrid_object);
+    delete hybrid_object;
+    hybrid_object = nullptr;
```

Comment:
Ok. Finally I got something that proved my concern here.

The native crash on GC is still there, please check my log information:
07-06 14:38:20.875   358   358 I DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
07-06 14:38:20.875   358   358 I DEBUG   : Build fingerprint: 'samsung/trltexx/trlte:5.0.1/LRX22C/N910FXXE1BOD2:eng/test-keys'
07-06 14:38:20.875   358   358 I DEBUG   : Revision: '12'
07-06 14:38:20.875   358   358 I DEBUG   : ABI: 'arm'
07-06 14:38:20.875   358   358 I DEBUG   : pid: 21628, tid: 21686, name: GVRF Finalize T  >>> org.gearvrf.gvroutlinesample                                                 <<<
07-06 14:38:20.875   358   358 I DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
07-06 14:38:20.915   358   358 I DEBUG   :     r0 af254dc0  r1 a048fadc  r2 af254dc0  r3 00000000
07-06 14:38:20.915   358   358 I DEBUG   :     r4 000002b6  r5 76efe7d0  r6 76efe740  r7 00000000
07-06 14:38:20.915   358   358 I DEBUG   :     r8 32c23e80  r9 b4e0e400  sl af254dc0  fp ffffffff
07-06 14:38:20.915   358   358 I DEBUG   :     ip b3bc9a3c  sp a048fac8  lr b3bc9a54  pc 00000000  cpsr a00d0010
07-06 14:38:20.915   358   358 I DEBUG   : 
07-06 14:38:20.915   358   358 I DEBUG   : backtrace:
07-06 14:38:20.915   358   358 I DEBUG   :     #00 pc 00000000  <unknown>
07-06 14:38:20.915   358   358 I DEBUG   :     #01 pc 000c4a50  /data/app/org.gearvrf.gvroutlinesample-2/lib/arm/libgvrf.so (Java_org_gearvrf_NativeHybridObject_delete+20)
07-06 14:38:20.915   358   358 I DEBUG   :     #02 pc 0008e339  /data/dalvik-cache/arm/data@app@org.gearvrf.gvroutlinesample-2@base.apk@classes.dex
07-06 14:38:22.765   358   358 I DEBUG   : 
07-06 14:38:22.765   358   358 I DEBUG   : Tombstone written to: /data/tombstones/tombstone_02


Actually as far as I can see "hybrid_object = nullptr" may not work as you expected since hybrid_object is just a copy of the native pointer from Java side(because primitive type parameter will always be passed by value not reference.) It means , probably, it may be the case that there are more than one java objects are holding the same native pointer and it finally leads to this kind of duplicate removal of dangling pointer. I will try to further evaluate this problem and see if I can reach to a conclusion later,